sympy/series/limits.py

```diff
@@ -162,8 +162,12 @@ def doit(self, **hints):
                                         for m in Mul.make_args(a))
                                     for a in Add.make_args(w)))
                 if all(ok(w) for w in e.as_numer_denom()):
-                    u = Dummy(positive=(z0 is S.Infinity))
-                    inve = e.subs(z, 1/u)
+                    if z0 is S.NegativeInfinity:
+                        u = Dummy(positive=True)
+                        inve = e.subs(z, -1/u)
+                    else:
+                        u = Dummy(positive=(z0 is S.Infinity))
```

Comment:
Is the condition `z0 is S.Infinity` necessary? We know that `abs(z0) == oo`. It seems that if `z0` is not `-oo`, then it should be `oo`.